index.js

```diff
@@ -96,12 +94,10 @@ class ClientLogin extends EventEmitter {
 
   verify (response) {
     try {
-      this.response = JSON.parse(this.handshake.recv(response))
+      return JSON.parse(this.handshake.recv(response))
     } catch (err) {
-      this.emit('error', err)
-      return
+      return { err }
     }
```

Comment:
before dropping the eventemitter, do you have a simple test case or gist for webpack so i can see whats going on?

Reply:
Sure I will share the repo in a minute.

Reply:
EventEmitter works fine.